### PR TITLE
Fix #1590: Enable listing directories

### LINKED ISSUE #1590: No way to list all directories
There appears to be no way to list all directories for storage.  It's possible to list files, accounts, and shares but not directories.

### PR DESCRIPTION
**Pull request by troydai**

### CODE REVIEW

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def transform_file_output(result):
     more clearly distinguishes between files and directories. """
     new_result = []
 
-    for item in result.get('items', [result]):
+    iterable = result if isinstance(result, list) else result.get('items', result)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py]: What is the reason for this change?

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py [hunk 2]
```diff
@@ -139,6 +139,16 @@
...
 """
 
+helps['storage directory list'] = """
+    type: command
+    short-summary: List directories in the specified share.
+    parameters:
+        - name: --num-results
+          type: integer
+          short-summary: Specifies the maximum number of directories to return. The default value is
+                         5000. The value cannot be less or equal to zero.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py]: If this is used as the input to the SDK call to get files and directories, then this is not accurate help text because any files (which will be filtered out) would count against the 5000 count quota.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py [hunk 3]
```diff
@@ -163,6 +173,10 @@
 helps['storage file list'] = """
     type: command
     short-summary: List files and directories in the specified share.
+    parameters:
+        - name: --files-only
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py]: Consider something like '--omit-directories' or '--suppress-directories'. For '--file-only' I would need to read the help text to know what that means. The alternative are (at least to mean) a little more self-explanatory.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py (hunk 1 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py]: Returning value from the custom command could be a list instead of a `ListGenerator`. That is case when directories are filtered.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py (hunk 2 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py]: Good point. Then this parameter cannot be accurately functional.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py (hunk 3 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py]: ok

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py (hunk 2 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py]: Arguably you could simply omit it.

Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_help.py]: Do you like the idea to hide the option on both `az storage file list` and `az storage directory list` until we have a good paging story?
